pkg/operator/ceph/cluster/controller.go

```diff
@@ -461,23 +462,50 @@ func (c *ClusterController) checkPVPresentInCluster(drivers []string, clusterID
 	return false, nil
 }
 
-// removeFinalizer removes a finalizer
-func removeFinalizer(client client.Client, name types.NamespacedName) error {
-	cephCluster := &cephv1.CephCluster{}
-	err := client.Get(context.TODO(), name, cephCluster)
+func removeFinalizers(client client.Client, name types.NamespacedName) error {
+	// Remove cephcluster finalizer
+	err := removeFinalizer(client, name, &cephv1.CephCluster{}, "")
+	if err != nil {
+		return errors.Wrap(err, "failed to remove cephcluster finalizer")
+	}
+
+	// Remove finalizer for rook-ceph-mon secret
+	name = types.NamespacedName{Name: mon.AppName, Namespace: name.Namespace}
+	err = removeFinalizer(client, name, &corev1.Secret{}, mon.DisasterProtectionFinalizerName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to remove finalizer for the secret %q", name.Name)
+	}
+
+	// Remove finalizer for rook-ceph-mon-endpoints configmap
+	name = types.NamespacedName{Name: mon.EndpointConfigMapName, Namespace: name.Namespace}
+	err = removeFinalizer(client, name, &corev1.ConfigMap{}, mon.DisasterProtectionFinalizerName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to remove finalizer for the configmap %q", name.Name)
+	}
+	return nil
+}
+
+func removeFinalizer(client client.Client, name types.NamespacedName, obj client.Object, finalizer string) error {
+	err := client.Get(context.TODO(), name, obj)
```

Comment:
Do not use `context.TODO()` there is a parent context in this file already: `OpManagerCtx` or just require a context in the function's signature.

Reply:
Done